src/catalog/models/base_model.py

```diff
@@ -0,0 +1,111 @@
+# coding=utf-8
+# Copyright 2018-2020 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from sqlalchemy import Column, Integer
+from sqlalchemy.exc import DatabaseError
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy_utils import database_exists, create_database, drop_database
+
+from src.catalog.sql_config import SQLConfig
+from src.utils.logging_manager import LoggingLevel
+from src.utils.logging_manager import LoggingManager
+
+db_session = SQLConfig().session
+
+
+class CustomModel:
+    """This overrides the default `_declarative_constructor` constructor.
+
+    It skips the attributes that are not present for the model, thus if a
+    dict is passed with some unknown attributes for the model on creation,
+    it won't complain for `unkwnown field`s.
+    Declares and int id field for all tables
+    """
+    query = db_session.query_property()
+    _id = Column('id', Integer, primary_key=True)
+
+    def __init__(self, **kwargs):
```

Comment:
Can we reuse the db_session functions across different catalog objects (e.g., datasets, columns, udfs, etc.)?

Reply:
Yes